src/main/java/li/cil/tis3d/common/module/SequencerModule.java

```diff
@@ -124,46 +129,44 @@ public void onData(final ByteBuf data) {
         }
     }
 
+    // XXX I'm giving up on this module for now. Transparency is busted,
+    // but should still be usable
     @Environment(EnvType.CLIENT)
     @Override
-    public void render(final BlockEntityRenderDispatcher rendererDispatcher, final float partialTicks) {
+    public void render(final BlockEntityRenderDispatcher rendererDispatcher, final float partialTicks,
+                       final MatrixStack matrices, final VertexConsumerProvider vcp,
+                       final int light, final int overlay) {
         if (!isVisible()) {
             return;
         }
 
-        rotateForRendering();
-        RenderUtil.ignoreLighting();
-        GlStateManager.enableBlend();
-
-        GlStateManager.disableTexture();
-        GlStateManager.depthMask(false);
+        matrices.push();
+        rotateForRendering(matrices);
 
+        final VertexConsumer vcCutout = vcp.getBuffer(RenderLayer.getCutoutMipped());
+        final RenderLayer translLayer = RenderLayer.getTranslucent();
+        final VertexConsumer vcTransl = vcp.getBuffer(translLayer);
+        final MatrixStack.Entry mat = matrices.peek();
         final boolean enabled = getCasing().isEnabled();
-        if (enabled) {
```

Comment:
It's because of the relative `matrices.translate()` calls to push the current Z position forward, for the render order issue I discussed. I'll try to see if I can rephrase the render code to produce less changes though.  
However as I mentioned in the comment above, the ported visual appearance doesn't look good regardless; transparency is my biggest problem. I don't know why but the render layers offering blending always make the quads turn solid when viewed at an angle not close to 90° (you can see this yourself by moving up very close to the module and watching the rectangle pass by). This is probably lighting-related, I should try using a custom render layer with blending and no lighting, now that I know how to do it.

Reply:
OK, cool. Thanks for the explanation!